File: tags/2.1.1/src/it/grabz/grabzit/GrabzItException.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package it.grabz.grabzit;

import it.grabz.grabzit.enums.ErrorCode;

/**
 *
 * @author GrabzIt
 */
public class GrabzItException extends Exception {
    private ErrorCode code;

    public GrabzItException(String message, ErrorCode code)
    {        
        super(message);
        this.code = code;
    }
    
    public GrabzItException(String message, String code)
    {
        this(message, ErrorCode.values()[Integer.parseInt(code)]);
    }
    /**
     * @return the code
     */
    public ErrorCode getCode() {
        return code;
    }
    
    
}

File: tags/php/2.1.5/aspnet/GrabzIt/GrabzItException.cs
using System;
using GrabzIt.Enums;

namespace GrabzIt
{
    public class GrabzItException : Exception
    {
        public ErrorCode code;

        internal GrabzItException(string message, string code)
            : this(message, (ErrorCode)Convert.ToInt32(code))
        {
        }

        internal GrabzItException(string message, ErrorCode code)
            : base(message)
        {
            this.code = code;
        }

        public ErrorCode Code
        {
            get
            {
                return code;
            }
            set
            {
                Code = code;
            }
        }
    }
}
